### PR TITLE
[LMS-664] [MARKUP] Trainee List page

### PR DESCRIPTION
**Pull request by FranzDelosSantos**
## Issue Link
https://framgiaph.backlog.com/board/LMS?selectedIssueKey=LMS-664&assignee=417519

## Defintion of Done
- [x] display list of trainees in a table, refer to the attachments
- [x] list of trainees should be displayed and correctly aligned ( First Name , Last Name , Role , Actions)
- [x] pagination displayed correctly and has total numbers on the list page also it has next button and previous button

## Steps to reproduce
docker--compose up

## Affected Components / Functionalities / Page
pages/trainer/trainees/index.tsx
shared/components/table/index.tsx
shared/icons/EyeIcon.tsx

## Notes
The pagination and search has no functionality yet based on the backlog.
Created a Reusable table component
Renamed the old table that is not used so that there is no error running npm run build since I created a new table.
## Screenshots
image

### CODE REVIEW

File: client/src/shared/components/Table/TableOld.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  };
+
+  return (
+    <div className="overflow-auto text-neutral-900 flex justify-center rounded-[5px]">
+      <table className="text-left w-full">
+        <thead className="bg-neutral-200">
+          <tr>
+            <th className="p-4">
+              {checkbox && <input type="checkbox" className="h-5 w-5 hover:bg-sky-700" />}
+            </th>
```
Comment (nicoleamber) [inline client/src/shared/components/Table/TableOld.tsx:53]: Instead of creating another table component, I think you can modify the old one. 
You can add a prop here to make the header dynamic (would rows have a column for checkbox or not):

```suggestion
            {selectableRows && <th className="p-4">
              {checkbox && <input type="checkbox" className="h-5 w-5 hover:bg-sky-700" />}
            </th>
            }
```

Just set a default to that prop in this file and adjust it accordingly on the component you're using it on.

File: client/src/shared/components/Table/index.tsx [hunk 2]
```diff
@@ -1,73 +1,61 @@
...
           </tr>
         </thead>
-        <tbody>{children}</tbody>
+        <tbody>
+          {data.map((item) => (
+            <tr
+              className={`${item.id % 2 === 0 ? '' : 'bg-gray-100'} align-middle cursor-pointer `}
+              key={item.id}
+              onClick={onClick}
+            >
```
Comment (nicoleamber) [inline client/src/shared/components/Table/index.tsx:36]: You can make use of `shared/components/Table/ListItem` component for the table rows since it already contains a column for `Actions`. 

You can just modify the row to be clickable with a prop i.e:
```suggestion
            <tr
              className={`${isClickable && 'cursor-pointer'} `}
              key={item.id}
              onClick={isClickable? onClick : undefined}
            >
```

File: client/src/pages/element/material/index.tsx
Comment (nicoleamber) [inline client/src/pages/element/material/index.tsx:1]: Is this file needed for the task? I think it was deleted in the recent updates.

File: client/src/shared/components/Table/index.tsx [hunk 3]
```diff
@@ -44,15 +44,17 @@ const Table: React.FC<TableProps> = ({
...
+        <thead>
           <tr>
-            <th className="p-4">
-              {checkbox && <input type="checkbox" className="h-5 w-5 hover:bg-sky-700" />}
-            </th>
+            {checkbox && (
+              <th className="bg-lightGray2 h-[49px]">
+                <input type="checkbox" className="h-5 w-5 hover:bg-sky-700" />
+              </th>
+            )}
```
Comment (nicoleamber) [inline client/src/shared/components/Table/index.tsx:55]: This causes a broken design for tables with checkboxes on rows but none on columns. (see image)
image


```suggestion
            {selectableRows && (
              <th className="bg-lightGray2 h-[49px]">
              {checkbox && <input type="checkbox" className="h-5 w-5 hover:bg-sky-700" />}
              </th>
          )}
```

add `selectableRows?: boolean` to TableProps and pass default into Table: `selectableRows = true`

File: client/src/shared/components/Table/ListItem.tsx [hunk 4]
```diff
@@ -31,10 +32,14 @@ export const ListItem: any = <T extends Data>({
...
   onCheckboxChange,
+  isClickable,
   register,
 }: ListItemProps<T>) => {
   return (
-    <tr className="even:bg-neutral-50 whitespace-nowrap text-sm text-black1 font-sans h-5 group transition-all ease-out duration-200">
+    <tr
+      className="even:bg-neutral-50 whitespace-nowrap text-[13px] text-lightGray3 font-sans h-[49px] group transition-all ease-out duration-200 cursor-pointer"
+      onClick={isClickable}
+    >
```
Comment (nicoleamber) [inline client/src/shared/components/Table/ListItem.tsx:42]: This will show `cursor-pointer` on all table rows including those that aren't clickable. You can refer to my previous comment: https://github.com/framgia/sph-lms/pull/133#discussion_r1230384928

**Review by nicoleamber — APPROVED** [review]

**Comment by nicoleamber** [comment]
Iterations: 2
Comments: 5
Testing Time: 30mins

**Review by Melanthonyando — APPROVED** [review]
### Approved

Iteration : 0
Comment: 0
Testing Time : 20 mins.
